### PR TITLE
feat: currency selector

### PR DESCRIPTION
**Pull request by JackShort**
* Adds currency selector and active currency
* does not currently have any effect other than the UI
* adds currency based off url params
* note 4 of the currencies don't have logos as i am waiting on design

figma link: https://www.figma.com/file/MOvvpkzygJdxCA3BY9mN1O/Mini-portfolio-v1?type=design&node-id=1861%3A211436&mode=design&t=4yX3eygA7IIxg7Zc-1

_the pr is not as large as it looks - a lot of the additions come from the svgs_

Screen Shot 2023-08-21 at 3 47 47 PM

Screen Shot 2023-08-21 at 3 47 54 PM

### CODE REVIEW

**Review by zzmp — DISMISSED** [review]
If there's no functional change, can you hide this behind a flag so it's not distracting to normal users?

**Comment by JackShort** [comment]
> If there's no functional change, can you hide this behind a flag so it's not distracting to normal users?

it is actually already hidden behind a feature flag @zzmp

**Review by cartcrom — COMMENTED** [review]
For variable/component/type names, could we consider using a convention like `FiatCurrency`, `PaperCurrency`, or `LocalCurrency` instead of just `Currency`, to avoid confusion w/ the Currency type from sdk core?

**Comment by JackShort** [comment]
> For variable/component/type names, could we consider using a convention like `FiatCurrency`, `PaperCurrency`, or `LocalCurrency` instead of just `Currency`, to avoid confusion w/ the Currency type from sdk core?

Good call out that completely went over my head

**Review by cartcrom — COMMENTED** [review]
also the right-aligned text looks a little big/unbalanced

 image

**Comment by JackShort** [comment]
> also the right-aligned text looks a little big/unbalanced
> 
> 
> 
>  image
> 
> 

That's actually the design

**Comment by cartcrom** [comment]
> > also the right-aligned text looks a little big/unbalanced
> >  image
> 
> That's actually the design

@infredible @bryankuuu i think the size diff is a little jarring here. but i might be crazy

File: src/hooks/useLocalActiveCurrency.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  if (typeof parsedLocalCurrency !== 'string') return undefined
+
+  const lowerCaseSupportedLocalCurrency = parsedLocalCurrency.toLowerCase()
+  return SUPPORTED_LOCAL_CURRENCIES.find(
+    (localCurrency) => localCurrency.toLowerCase() === lowerCaseSupportedLocalCurrency
+  )
+}
+
+export function useActiveLocalCurrency(): SupportedLocalCurrency {
```
Comment (cartcrom) [inline src/hooks/useLocalActiveCurrency.ts:24]: 🚨 File name is different than hook name 🚔

**Review by cartcrom — APPROVED** [review]
1 filename nit that would be good to fix, but LGTM!!

File: src/hooks/useLocalActiveCurrency.ts (hunk 1 above)
Comment (JackShort) [reply src/hooks/useLocalActiveCurrency.ts:24]: thats cringe

File: src/components/AccountDrawer/DefaultMenu.tsx [hunk 2]
```diff
@@ -48,9 +51,14 @@ function DefaultMenu({ drawerOpen }: { drawerOpen: boolean }) {
...
       {menu === MenuState.SETTINGS && (
-        <SettingsMenu onClose={closeSettings} openLanguageSettings={openLanguageSettings} />
+        <SettingsMenu
+          onClose={closeSettings}
+          openLanguageSettings={openLanguageSettings}
+          openLocalCurrencySettings={openLocalCurrencySettings}
+        />
       )}
       {menu === MenuState.LANGUAGE_SETTINGS && <LanguageMenu onClose={openSettings} />}
+      {menu === MenuState.LOCAL_CURRENCY_SETTINGS && <LocalCurrencyMenu onClose={openSettings} />}
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/components/AccountDrawer/DefaultMenu.tsx:61]: you can't do a switch statement inside of a rendering function - in order to do so I would have to take everything into another function and considering the rendering function only wraps the options in one div that doesn't make too much sense to do

File: src/constants/localCurrencies.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  RUB_ICON,
+  SGD_ICON,
+  THB_ICON,
+  TRY_ICON,
+  UAH_ICON,
+  USD_ICON,
+  VND_ICON,
+} from './localCurrencyIcons'
+
+export const SUPPORTED_LOCAL_CURRENCIES = [
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/constants/localCurrencies.tsx:24]: following what we did with languages and just doing alphabetical - USD

File: src/components/AccountDrawer/DefaultMenu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/components/AccountDrawer/DefaultMenu.tsx:61]: You could do something like this in the main body of the component. It's also fairly simple to pull that out into it's own `getMenuContent` helper function if that's preferred. 
```
  let menuContent: JSX.Element
  switch (menu) {
    case MenuState.DEFAULT:
      menuContent = isAuthenticated ? (
        <AuthenticatedHeader account={account} openSettings={openSettings} />
      ) : (
        <WalletModal openSettings={openSettings} />
      )
      break
    case MenuState.SETTINGS:
      menuContent = (
        <SettingsMenu
          onClose={closeSettings}
          openLanguageSettings={openLanguageSettings}
          openLocalCurrencySettings={openLocalCurrencySettings}
        />
      )
      break
    case MenuState.LANGUAGE_SETTINGS:
      menuContent = <LanguageMenu onClose={openSettings} />
      break
    case MenuState.LOCAL_CURRENCY_SETTINGS:
      menuContent = <LocalCurrencyMenu onClose={openSettings} />
      break
    default:
      menuContent = <WalletModal openSettings={openSettings} />
  }

  return <DefaultMenuWrap>{menuContent}</DefaultMenuWrap>
```

**Review by cbachmeier — APPROVED** [review]
LGTM, but check out my comment on switch statement options

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**